File: tests/fixtures/polyglot/api/lib/pricing.ts
export const PREMIUM_DISCOUNT = 0.1;

export function baseTotal(items) {
  return items.reduce((sum, item) => sum + item.price * item.quantity, 0);
}

export function priceFor(items, tier = 'standard') {
  const total = baseTotal(items);
  return tier === 'premium' ? total * (1 - PREMIUM_DISCOUNT) : total;
}

File: tests/fixtures/polyglot/core/pricing.py
"""Pricing rules."""
import os

PREMIUM_DISCOUNT = 0.10
TAX_RATE = float(os.environ.get('TAX_RATE', '0.15'))


def base_total(items):
    return sum(item['price'] * item['quantity'] for item in items)


def price_for(items, tier='standard'):
    total = base_total(items)
    if tier == 'premium':
        total = total * (1 - PREMIUM_DISCOUNT)
    return round(total * (1 + TAX_RATE), 2)
